### PR TITLE
Added num_gpus behavior for TabTransformer.

### PR DESCRIPTION
**Pull request by Chudbrochil**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-793-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-793/1/index.html

**Comment by Chudbrochil** [comment]
Also updated default hyperparameters to defaults found while doing HPO.

File: tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py [hunk 1]
```diff
@@ -47,11 +47,19 @@ def __init__(self, **kwargs):
         self._temp_file_name = "tab_trans_temp.pth"
         self._period_columns_mapping = None
 
-        # TODO: Take in num_gpu's as a param. Currently this is hard-coded upon detection of cuda.
-        if torch.cuda.is_available():
-            self.device = torch.device("cuda")
-        else:
+        num_gpus = self.params.get('num_gpus', None)
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py:50]: `num_gpus` is not in `params`, it is a kwarg of `_fit()`. See here for example:

https://github.com/awslabs/autogluon/blob/master/tabular/src/autogluon/tabular/models/tabular_nn/tabular_nn_model.py#L194

File: tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py [hunk 2]
```diff
@@ -527,6 +535,7 @@ def load(cls, path: str, reset_paths=False, verbose=True):
         
         3) Bug where HPO doesn't work when cuda is enabled.
         "RuntimeError: Cannot re-initialize CUDA in forked subprocess. To use CUDA with multiprocessing, you must use the 'spawn' start method"
+        Update: This will likely be fixed in a future change to HPO in AutoGluon.
         
         4) Enable output layer of TT model to be multiple fully connected layers rather than just a single
```
Comment (jwmueller) [inline tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py:540]: since 4) is completed, you can remove this :)

**Review by jwmueller — CHANGES_REQUESTED** [review]

File: tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py (hunk 2 above)
Comment (Chudbrochil) [reply tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py]: Got it!

File: tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py (hunk 1 above)
Comment (Chudbrochil) [reply tabular/src/autogluon/tabular/models/tab_transformer/tab_transformer_model.py]: Moved to _fit and changed self.params to kwargs.

**Comment by szha** [comment]
Job PR-793-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-793/4/index.html

**Review by jwmueller — APPROVED** [review]
looks good

**Pull request merged**